lib/analytics.js

```diff
@@ -0,0 +1,115 @@
+const fs = require('fs');
+
+// We can either use a process-wide object regularly saved to a JSON file,
+// or a Redis equivalent (for multi-process / when the filesystem is unreliable.
+var redis;
+var useRedis = false;
```

Comment:
@paulmelnikow Yeah, sorry, I didn't realise initially, and added this comment later:
> I wrote a few of these comments not realising that you were simply moving code around. My mistake, but the comments might still be useful for future cleanup.

 GitHub doesn't point out that code has simply been moved 😢  Phabricator does a really good job of that - It has special highlighting for lines of code that were moved.